### ISSUE TITLE
PlotHistogram with custom functionality

### ISSUE THREAD

**Issue by q-depot** [ocornut/imgui#1157]
Hi,

I'd like to make a custom widget based on the PlotHistogram, basically I want to add an editable gain curve on top. The mouse left click would create a control point, right click delete it and drag would move it within the frame.

This shouldn't be too difficult but I was wandering what's the best way to implement a custom widget and how to capture/implement the mouse events. I would also need to remove the PlotHistogram overlay.

thanks.

**Comment by ocornut** [comment]
Hello @q-depot, have you solved your problem?

Right now the typical way would be to copy the Plot() functions and just make your own based on that (add them in your own .cpp file by including `imgui_internals.h`). However the problem is that forward compatibility isn't guaranteed. Although the internals don't change very often they may change from time to time.

In the future I would hope to introduce a more fully featured plotting/graphing/charting extension.

Some other references:
https://github.com/ocornut/imgui/issues/786
https://github.com/ocornut/imgui/issues/632

**Comment by q-depot** [comment]
Hi,

yes that's what I did more or less. I've also managed to implement the gain curve although I'm not sure about performance, I followed [this](https://github.com/ocornut/imgui/issues/786) bezier widget as example, so for each control point I create an invisible button. 
I thought I could be smarted and create only 1 button for the whole widget and then check the mouse position, but then when the cursors goes out of the canvas while dragging a point, the UX is weird.

thanks
